vmdb/app/controllers/host_controller.rb

```diff
@@ -341,7 +341,9 @@ def form_field_changed
     render :update do |page|                    # Use JS to update the display
       if changed != session[:changed]
         session[:changed] = changed
-        page << javascript_for_miq_button_visibility(changed)
+        unless %w(new).include?(params[:id])
```

Comment:
@romanblanco could have just checked for value like unless params[:id] == "new", params[:id] will either be an id of existing record or "new"